File: frontend/lib/services/analysis.ts
import { fetchAPI } from '@/lib/api-client';

export interface Analysis {
  id: string;
  brand_id?: string;
  analysis_type: string;
  input_data: any;
  results?: any;
  score?: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  ai_model?: string;
  error_message?: string;
  created_at: string;
  completed_at?: string;
  avg_position?: number;
  inclusion_rate?: number;
}

export interface CreateAnalysisData {
  brand_id?: string;
  analysis_type: string;
  input_data: {
    domain?: string;
    content?: string;
    [key: string]: any;
  };
  ai_model?: string;
}

export const analysisService = {
  getAll: async (brandId?: string) => {
    const query = brandId ? `?brand_id=${brandId}` : '';
    return fetchAPI<Analysis[]>(`/analysis/${query}`);
  },

  getById: async (id: string) => {
    return fetchAPI<Analysis>(`/analysis/${id}`);
  },

  create: async (data: CreateAnalysisData) => {
    return fetchAPI<Analysis>('/analysis/', {
      method: 'POST',
      body: JSON.stringify(data),
    });
  },
};

File: backend/app/models/analysis.py
from pydantic import BaseModel, Json
from typing import Optional, Any, Dict
from uuid import UUID
from datetime import datetime
from enum import Enum

class AnalysisType(str, Enum):
    domain = 'domain'
    content = 'content'
    keyword = 'keyword'
    competitor = 'competitor'

class AIModel(str, Enum):
    chatgpt = 'chatgpt'
    claude = 'claude'
    perplexity = 'perplexity'
    gemini = 'gemini'
    openrouter = 'openrouter'

class AnalysisStatus(str, Enum):
    pending = 'pending'
    processing = 'processing'
    completed = 'completed'
    failed = 'failed'

class AnalysisBase(BaseModel):
    brand_id: Optional[UUID] = None
    analysis_type: AnalysisType
    input_data: Dict[str, Any]
    ai_model: Optional[AIModel] = None
    model_name: Optional[str] = None # e.g. "gpt-4", "anthropic/claude-3-opus"

class AnalysisCreate(AnalysisBase):
    pass

class AnalysisUpdate(BaseModel):
    results: Optional[Dict[str, Any]] = None
    score: Optional[float] = None
    status: Optional[AnalysisStatus] = None
    error_message: Optional[str] = None
    completed_at: Optional[datetime] = None

class Analysis(AnalysisBase):
    id: UUID
    user_id: UUID
    results: Optional[Dict[str, Any]] = None
    score: Optional[float] = None
    status: AnalysisStatus
    error_message: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True
